datasource/ethereum/src/block_ingestor.rs

```diff
@@ -262,9 +296,25 @@ where
             batching_web3
                 .transport()
                 .submit_batch()
-                .map_err(|e| format_err!("could not get blocks from Ethereum: {}", e))
+                .map_err(|e| format_err!("could not get blocks from Ethereum: {}", e).into())
                 .map(|_| stream::futures_unordered(block_futures))
                 .flatten_stream(),
         )
     }
 }
+
+#[derive(Debug)]
+enum BlockIngestorError {
```

Comment:
Even though you chose to convert from `Error` to this rather than the other way around, it's still good to derive `Fail` so we get `Display`.

Reply:
Oh good idea. Will do